In [1]:
import pandas as pd
import statsmodels.api as sm
import scipy.stats as stats
import pingouin as pg
import seaborn as sns

In [2]:
# доступ к файлу удалось получить только предварительно скачав его на ПК
file = '/Users/Administrator/Documents/Finale Files/titanic3.xls'
df0 = pd.read_excel(file)

In [3]:
df = df0[['age', 'survived']].dropna()

In [4]:
#scipy
survived = df[df['survived'] == 1]['age']
not_survived= df[df['survived'] == 0]['age']
t_statistic, p_value = stats.ttest_ind(survived, not_survived)
f"results_scipy: T-статистика: {t_statistic:.4f}, P-значение: {p_value:.4f}"

'results_scipy: T-статистика: -1.7964, P-значение: 0.0727'

In [5]:
#pingouin
survived = df[df['survived'] == 1]['age']
not_survived= df[df['survived'] == 0]['age']
results_pingouin = pg.ttest(survived, not_survived)[['T', 'p-val']]
results_pingouin

T     p-val
T-test -1.770718  0.076958

Вывод: Поскольку P-значение: 0.0727 > 0.05 (уровень ошибки I рода) - принимаем нуль-гипотезу, что средний возраст спасенных пассажиров равен среднему возрасту погибших пассажиров.